solr/api/src/java/org/apache/solr/client/api/model/InstallShardDataRequestBody.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.client.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class InstallShardDataRequestBody {
+  @JsonProperty(value = "location", required = true)
```

Comment:
It wouldn't surprise me if there were a few that were missed: this PR is largely a refactor of existing classes written by multiple folks, so I'm sure there's some inconsistency there.

I'm reluctant to get too down into the weeds of policing these annotation values at this point as it's not the aim of my current ticket, and because there are other tickets that I think will be a more comprehensive forcing function for that. (e.g. when we start using some of the generated v2 SolrRequest implementations in our tests).

But `required` is probably important enough to act on here - thanks for pointing this out